src/history.cpp

```diff
@@ -1453,7 +1453,10 @@ bool history_t::save_internal_via_rewrite()
                 if (wstat(new_name, &sbuf) >= 0)
                 {
                    /* Success */
-                    fchown(out_fd, sbuf.st_uid, sbuf.st_gid);
+                    if (0 > fchown(out_fd, sbuf.st_uid, sbuf.st_gid))
```

Comment:
I'm willing to bet whomever wrote that did so based on a more defensible idiom: `if (0 == x)`. That way if one of the equal-signs was omitted the compiler will complain about assigning to a constant which `if (x = 0)` would not do and result in a bug.  But even that idiom is not justified these days given modern compilers and linters.